Exposing Whitelisted Protocols to User Settings

jonassorgenfrei opened the pull request:
**What's this PR do?**

It exposes the whitelisted protocols list so users can add additional protocols that should not use the link wrapper. 
Adresses Issue: #1284

**You have tested this PR on:**

- [x] Windows
- [x] Linux/Ubuntu

alya commented:
Thanks! Could you please update your commit message to match the [commit style guidelines](https://zulip.readthedocs.io/en/latest/contributing/version-control.html#commit-messages)?

andersk commented:
Do not close and reopen multiple pull requests for the same thing, even to fix the commit structure; you can always fix the commit structure by rebasing and force-pushing your existing branch.

jonassorgenfrei commented:
I adjusted the commit message to match the style guide. 
Yeah i messed i my repo on the first PR by directly implementing it on the main branch and by messing up my rebasing. Next time i know better :)

jonassorgenfrei commented:
Is there anything left to do on this PR?

jonassorgenfrei commented:
@andersk @timabbott can you take a look if there's anything missing here that avoid a merge?

timabbott commented:
I think we'd need documentation for how users can decide which protocols they can safely add there.

jonassorgenfrei commented:
@timabbott 
> I think we'd need documentation for how users can decide which protocols they can safely add there.

Totally makes sense, i'm just unsure where to document the setting.json 's new key. 
I could put it in the docs but would probably have to add a new .md file like desktop-app-settings.md? 
On Zulip Help i didn't find a section about the setting.json. 

Could you point me to where i shall add the information then i'm happy to add it there too

timabbott commented:
A new file seems fine. I'd call it `customize-link-protocols.md`

jonassorgenfrei commented:
Allright i added a customize-link-protocols.md to docs/howto :)
Hope that's fine.

jonassorgenfrei committed:
link_utils: exposing whitelisted protocols to user settings

Adding config option to set protocols in the config that are
whitelisted to be opened directly.
The behaviour is documented in docs\howto\customize-link-protocols.md.

jonassorgenfrei committed:
link_utils: exposing whitelisted protocols to user settings

Fixes #1284.
Adding config option to set protocols in the config that are
whitelisted to be opened directly.
The behaviour is documented in docs\howto\customize-link-protocols.md.

jonassorgenfrei commented:
@timabbott could you may be take another look on the PR? 

Everything should be there now, would be cool if we can merge it in main,
as i often had to fix merge conflicts with the latest changes to keep this pr uptodate.

shubham-padia commented on app/common/link-util.ts:
```diff
@@ -3,11 +3,21 @@ import fs from "node:fs";
 import {Html, html} from "./html.ts";
 import * as t from "./translation-util.ts";
 
+/* Fetches the current protocolLaunchers from settings.json */
```
protocolLaunchers doesn't seem like a noun. I don't think this comment is needed, since it is clear from using `ConfigUtil.getConfigItem(` that we are fetching from settings.json

shubham-padia commented on docs/howto/customize-link-protocols.md:
````diff
@@ -0,0 +1,35 @@
+By default, the following protocols are whitelisted:
+
+```
+http https mailto tel sip
````
Maybe we can separate this using a comma?

shubham-padia commented on docs/howto/customize-link-protocols.md:
```diff
@@ -0,0 +1,35 @@
+
+It is possible to customize the list of whitelisted protocols by editing the `settings.json` file located at: `userdata/Zulip/config/settings.json`
+
+To add or modify the list, the `whitelistedProtocols` key can be updated. For example:
```
`To add or modify` -> `To modify`

shubham-padia commented on docs/howto/customize-link-protocols.md:
```diff
@@ -0,0 +1,35 @@
+
+Links using these protocols are opened directly by the system.
+
+All other protocols are considered potentially unsafe and are therefore opened indirectly—through a local HTML file—in your default web browser.
```
`file-in` -> `file in`

shubham-padia reviewed (commented):
This is great work! Tested and it works fine. Requested some changes for comments.

shubham-padia commented:
Opened #1467 with the proposed changes while keeping the author same.

jonassorgenfrei commented:
Hey @shubham-padia 

Sorry I haven't checked back. But I saw you did the changes yourself already thanks for that. 

I will close this Mr then?

jonassorgenfrei commented:
Hey @shubham-padia 

I've adopted your changes and updated the PR.
Can you review again and merge if good?

Thanks a lot 

cheers 
Jonas
